server/types/resolvable.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/dolthub/vitess/go/vt/proto/query"
+
+	"github.com/dolthub/doltgresql/postgres/parser/sem/tree"
+)
+
+// ResolvableType represents any non-built-in type
+// that needs resolution at analyzer stage.
+// It is used for domain types, and it can be used
+// for other user-defined types we don't support yet.
```

Comment:
The plan is for this to be used for all types, essentially in place of returning `DoltgresType` in `server/ast/resolvable_type_reference.go::nodeResolvableTypeReference`